File: jv-service/src/main/java/com/lcaohoanq/jvservice/domain/mail/IMailService.java
package com.lcaohoanq.jvservice.domain.mail;

import com.lcaohoanq.jvservice.domain.user.User;
import io.reactivex.rxjava3.core.Single;
import jakarta.mail.MessagingException;
import org.thymeleaf.context.Context;

public interface IMailService {
    void sendMail(String to, String subject, String templateName, Context context) throws MessagingException;
    Single<User> createEmailVerification(User user);
}

File: notification-service/src/main/kotlin/com/lcaohoanq/notificationservice/domains/mail/IMailService.kt
package com.lcaohoanq.notificationservice.domains.mail

import feign.FeignException
import org.thymeleaf.context.Context

interface IMailService {
    fun sendMail(to: String, subject: String, templateName: String, context: Context)
    abstract fun ServiceUnavailable(s: String): FeignException.ServiceUnavailable
//    fun createEmailVerification(user: User): Single<User>
}
